python/ql/src/semmle/python/pointsto/PointsTo.qll

```diff
@@ -2091,18 +2091,23 @@ cached module Types {
 
 module AttributePointsTo {
 
+    pragma [noinline]
     predicate pointsTo(ControlFlowNode f, Context context, ObjectInternal value, ControlFlowNode origin) {
         exists(EssaVariable var, string name, CfgOrigin orig |
-            f.isLoad() and
-            var.getASourceUse() = f.(AttrNode).getObject(name)
-            or
-            Expressions::getattr_call(f, var.getASourceUse(), context, _, name)
-            |
+            getsVariableAttribute(f, var, name) and
             variableAttributePointsTo(var, context, name, value, orig) and
             origin = orig.asCfgNodeOrHere(f)
         )
     }
 
+    pragma [noinline]
+    private predicate getsVariableAttribute(ControlFlowNode f, EssaVariable var, string name) {
+        Expressions::getattr_call(f, var.getASourceUse(), _, _, name)
```

Comment:
No `context`?

Reply:
The `pointsTo` predicate above includes the context, so it's mostly redundant here. A syntactic check would probably be sufficient, but this is quick enough.